xwiki-platform-core/xwiki-platform-skin/xwiki-platform-skin-skinx/src/main/java/com/xpn/xwiki/web/sx/SxDocumentSource.java

```diff
@@ -138,7 +138,8 @@ public String getContent()
                     LESSResourceReference lessResourceReference =
                         lessResourceReferenceFactory.createReferenceForXObjectProperty(objectPropertyReference);
                     try {
-                        sxContent = lessCompiler.compile(lessResourceReference, true, (parse == 1), false);
+                        // We don't use cache here because it should be managed by the browser using cache policy.
+                        sxContent = lessCompiler.compile(lessResourceReference, true, (parse == 1), true);
```

Comment:
So you never use the LESS cache? Note that the LESS compiler is very CPU-consuming. If you never cache, LESS compiler will be used for each individual request. IMO, you will just make the wiki crash all the time.

See: https://jira.xwiki.org/browse/XWIKI-15406?focusedCommentId=99267&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-99267

Reply:
> The LESS compiler will not always be used if you enable the SSX cache as the browser will 
> not make the request (if it has previously cached the page).

If thousands of new users (with empty cache) come in the same time, you will have trouble. What I would do is to disable the LESS cache when the developer of the SSX set "no cache" at the browser policy, but still use it otherwise. This way, it is the responsibility of the developer to set the correct cache value.

> Also why would the wiki crash if the LESS cache is disable?

Having too much load on a server is a good way to make it crash.

>  I guess it depends on how many requests are being made.

Definitively.
> What I can do is to disable velocity compiled code being cached and instead compile the 
> velocity code each time the request is made instead of the whole LESS + velocity.

I don't think you can do that because velocity is executed before LESS.

Reply:
> What I would do is to disable the LESS cache when the developer of the SSX set "no cache" at the browser policy, but still use it otherwise

Done, indeed you suggestion is better as LESS cache will be used at least when the cache policy is not set to "no caching".